odl/space/ntuples.py

```diff
@@ -119,12 +119,15 @@ def element(self, inp=None, data_ptr=None):
                 return self.element_type(self, arr)
         else:
             if data_ptr is None:
-                arr = np.array(inp, copy=False, dtype=self.dtype, ndmin=1)
-                if arr.shape != (self.size,):
-                    raise ValueError('expected input shape {}, got {}.'
-                                     ''.format((self.size,), arr.shape))
+                if inp in self:
+                    return inp
+                else:
+                    arr = np.array(inp, copy=False, dtype=self.dtype, ndmin=1)
```

Comment:
I think it actually does, but it is dependent on many other things. I decided to add an if statement above in order to return the input as is instead of returning a view. This should preserve `a is b` type statements and perhaps help in some cases.